tests/src/unit-udt_macro.cpp

```diff
@@ -390,9 +367,9 @@ TEST_CASE_TEMPLATE("Serialization/deserialization of classes with 26 public/priv
 
         {
             T obj1;
-            nlohmann::json j1 = obj1; //via msgpack
-            std::vector<uint8_t> buf = nlohmann::json::to_msgpack(j1);
-            nlohmann::json j2 = nlohmann::json::from_msgpack(buf);
+            json_t j1 = obj1;  //via msgpack
```

Comment:
I don't know how much sense it makes to test the different formats here. To the extent that there are roundtrip differences, these are best tested elsewhere.
You're also missing BJData, and when the BON8 PR is merged, this is likely to be missed.

Reply:
This is existing code, which I modified (parameterized on different types) to support different underlying implementations for json: ordered and unordered map

I'd prefer to tackle the missing BJData and BON8 in a separate PR. Thoughts?

Reply:
Here's the PR where they were added: https://github.com/nlohmann/json/pull/2287

Reply:
I just find the choice of roundtripping bizarre. It's already tested extensively for each format and unless you're testing the corner cases (which the macros don't handle anyway), there's no reason to do so here. Let's just do *something* to pad the unit test! :man_shrugging:

Reply:
Still, why test different formats? Ignoring format-specific exceptions, a roundtrip doesn't change the `basic_json`.

```cpp
SECTION("alphabet")
{
    json_t j = obj1;
    T obj2;
    j.get_to(obj2);
    CHECK((obj1 == obj2));
}
```

Is this not sufficient?

Reply:
Could be, don't know for sure.  @pfeatherstone did you have a particular reason for not just doing one format here?